src/ExpressiveInstaller/Resources/config/container-pimple.php

```diff
@@ -16,37 +17,41 @@
     && is_array($config['dependencies']['services'])
 ) {
     foreach ($config['dependencies']['services'] as $name => $service) {
-        $container[$name] = function ($c) use ($service) {
+        $container[$name] = function (Container $c) use ($service) {
             return $service;
         };
     }
 }
 
 // Inject factories
 foreach ($config['dependencies']['factories'] as $name => $object) {
-    $container[$name] = function ($c) use ($object, $name) {
-        if ($c->has($object)) {
-            $factory = $c->get($object);
+    $container[$name] = function (Container $c) use ($object, $name) {
+        $psrContainer = new PsrContainer($c);
```

Comment:
it was my concern here. Please note, that closure parameter will be normal Pimple Container and than we use `has` and `get` methods. I can use here instead of these methods `isset` and "direct" get (so exactly as it is in pimple `PsrContainer`) and later I think we need to pass to the factory `PsrContainer` anyway, so I think we need to create it here anyway...